lnd_test.go

```diff
@@ -387,6 +387,184 @@ func testBasicChannelFunding(net *networkHarness, t *harnessTest) {
+	assertTxInBlock(t, block, fundingTxID)
+	miner.Node.Generate(15)
+
+	// Ensure the chain lengths are what we expect.
+	_, newNodeHeight, err = miner.Node.GetBestBlock()
+	if err != nil {
+		t.Fatalf("unable to get current blockheight %v", err)
+	}
+
+	_, orgNodeHeight, err = net.Miner.Node.GetBestBlock()
+	if err != nil {
+		t.Fatalf("unable to get current blockheight %v", err)
+	}
+
+	if newNodeHeight != orgNodeHeight+5 {
+		t.Fatalf("expected new miner(%d) to be 5 blocks ahead of "+
+			"original miner(%d)", newNodeHeight, orgNodeHeight)
+	}
+
+	chanPoint := &lnrpc.ChannelPoint{
+		FundingTxid: pendingUpdate.Txid,
+		OutputIndex: pendingUpdate.OutputIndex,
+	}
+
+	// Ensure channel is no longer pending.
+	assertNumOpenChannelsPending(ctxt, t, net.Alice, net.Bob, 0)
+
+	// Wait for Alice and Bob to recognize and advertise the new channel
+	// generated above.
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err = net.Alice.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("alice didn't advertise channel before "+
+			"timeout: %v", err)
+	}
+	err = net.Bob.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("bob didn't advertise channel before "+
+			"timeout: %v", err)
+	}
+
+	// Alice should now have 1 edge in her graph.
+	req := &lnrpc.ChannelGraphRequest{}
+	chanGraph, err := net.Alice.DescribeGraph(ctxb, req)
+	if err != nil {
+		t.Fatalf("unable to query for alice's routing table: %v", err)
+	}
+
+	numEdges := len(chanGraph.Edges)
+	if numEdges != 1 {
+		t.Fatalf("expected to find one edge in the graph, found %d",
+			numEdges)
+	}
+
+	// Connecting the two miners should now cause our original one to sync
+	// to the new, and longer chain.
+	if err := rpctest.ConnectNode(net.Miner, miner); err != nil {
+		t.Fatalf("unable to connect harnesses: %v", err)
+	}
+
+	if err := rpctest.JoinNodes(nodeSlice, rpctest.Blocks); err != nil {
+		t.Fatalf("unable to join node on blocks: %v", err)
+	}
+
+	// Once again they should be on the same chain.
+	_, newNodeHeight, err = miner.Node.GetBestBlock()
+	if err != nil {
+		t.Fatalf("unable to get current blockheight %v", err)
+	}
+
+	_, orgNodeHeight, err = net.Miner.Node.GetBestBlock()
+	if err != nil {
+		t.Fatalf("unable to get current blockheight %v", err)
+	}
+
+	if newNodeHeight != orgNodeHeight {
+		t.Fatalf("expected new miner(%d) and original miner(%d) to "+
+			"be on the same height", newNodeHeight, orgNodeHeight)
+	}
+
+	// Since the fundingtx was reorged out, Alice should now have no edges
+	// in her graph.
+	req = &lnrpc.ChannelGraphRequest{}
+	chanGraph, err = net.Alice.DescribeGraph(ctxb, req)
+	if err != nil {
+		t.Fatalf("unable to query for alice's routing table: %v", err)
+	}
+
+	numEdges = len(chanGraph.Edges)
+	if numEdges != 0 {
+		t.Fatalf("expected to find no edge in the graph, found %d",
+			numEdges)
+	}
+
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	closeChannelAndAssert(ctxt, t, net, net.Alice, chanPoint, false)
```

Comment:
The `miner` instance created here should be shutdown as it's no longer used in any tests after this one (needed since there's also no code to remove _all_ the active btcd harnesses at the end of the integration test run as we do with all active lnd nodes).

Reply:
added a `defer miner.TearDown()` right after `SetUp`.